src/storage-client/src/controller/metrics_scraper.rs

```diff
@@ -0,0 +1,118 @@
+
+    mz_ore::task::spawn(|| "metrics_scraper", async move {
+        // Keep track of what we think the contents of the output
+        // collection, so that we can emit the required retractions/updates
+        // when we learn about new metrics.
+        //
+        // We assume that one update from the MetricsFetcher contains entries
+        // for all known provisioned instances and if instances are not in an
+        // update their updates should be removed from the output collection.
+        let mut current_metrics = ChangeBatch::new();
+
+        let mut interval = tokio::time::interval(METRICS_INTERVAL);
+
+        loop {
+            tokio::select! {
+                _msg = &mut shutdown_rx => {
+                    break;
+                }
+
+                _ = interval.tick() => {
+                    let metrics = metrics_fetcher.fetch_metrics().await;
+                    tracing::trace!("metrics: {:?}", metrics);
+
+                    let mut row_buf = Row::default();
+                    let mut correction = current_metrics
+                        .iter()
+                        .cloned()
+                        .map(|(row, diff)| (row, -diff))
+                        .collect_vec();
+
+                    // NOTE: host_addr is currently unused, it will be
+                    // logged at trace level, which might be useful when
+                    // debugging.
+                    for (service_id, _host_addr, metrics) in metrics {
+                        match metrics {
+                            Ok(process_metrics) => {
+                                for (process_id, metrics) in process_metrics.into_iter().enumerate()
+                                {
+                                    let mut packer = row_buf.packer();
+
+                                    packer.push(Datum::from(service_id.to_string().as_str()));
+                                    packer.push(Datum::from(u64::cast_from(process_id)));
+                                    packer.push(Datum::from(metrics.cpu_nano_cores));
+                                    packer.push(Datum::from(metrics.memory_bytes));
+
+                                    correction.push((row_buf.clone(), 1));
+                                }
+                            }
+                            Err(err) => {
+                                tracing::warn!(
+                                    "error fetching metrics for {service_id}: {:?}",
+                                    err
+                                );
+                            }
+                        }
+                    }
+
+                    // Update our view of the output collection and write updates
+                    // out to the collection.
+                    current_metrics.extend(correction.iter().cloned());
+                    collection_mgmt
+                        .append_to_collection(metrics_collection_id, correction)
+                        .await;
```

Comment:
we probably shouldnt send an update if its empty; the collection manager will do that for us each second